HW2

BoLjshOJ opened the pull request:
HW2 done

BoLjshOJ committed:
init
fix hw1 questions:
1. Для вопроса лучше выделить отдельную сущность и работать уже с ней
2. Функционал по собственно тестированию лучше вынести из пользователя в отдельный сервис, который с этим пользователем взаимодействует.
3. Т.к. файлы csv лежат в ресурсах, то и работать с ними лучше как с ресурсами (getClass().getClassLoader().getResourceAsStream(...))
4. FileDao можно сделать более универсальным. Интерфейсы для dao в т.ч. нужны для того, чтобы можно было легко подменить источник данных. Например с файла на БД. Поэтому лучше сделать что-то типа QuestionsDao с методом getQuestions(), а все что для этого нужно передать в конструкторе реализации FileQuestionsDaoImpl
5.По заданию csv файл один. Не принципиально, но в следующих заданиях может быть проще

BoLjshOJ committed:
HW 2
-Add constants in bundle
-Add _ru language
-Create MessageSource ability
-Porting app to cfg realization
-Update implements to MessageSource realization

BoLjshOJ committed:
HW 2
-Add test for example

stvort commented on src/main/java/ru/otus/springFramework/dao/QuestionDaoImpl.java:
```diff
@@ -0,0 +1,57 @@
+    public void initQuestionsFromCSV(){
+        this.questions = new ArrayList<>();
+        try {
+            InputStream inputStream = this.getClass().getClassLoader().getResourceAsStream(messageService.getMessage("pathForQuestions"));
```
Имя файла с вопросами все же не message.  Bundle часть системы, а файл с вопросами сменный источник данных. Лучше формировать имя файла из шаблона и локали. Можно сделать что-то типа бина, который хранит настройки и везде работать с ним. Он также может отвечать за формирование файла с вопросами

stvort commented on src/main/java/ru/otus/springFramework/dao/StudentDaoSimple.java:
```diff
@@ -0,0 +1,26 @@
+    }
+
+    public Student createStudent() {
+        Scanner scanner = new Scanner(System.in);
```
Работу с консолью (ввод/вывод) можно вынести в отдельный сервис. Так будет удобнее тестировать классы, где этот ввод/вывод используется

stvort commented on src/main/java/ru/otus/springFramework/config/AppConfig.java:
```diff
@@ -0,0 +1,24 @@
+    @Bean
+    public MessageSource messageSource() {
+        ReloadableResourceBundleMessageSource ms = new ReloadableResourceBundleMessageSource();
+        ms.setBasename("classpath:/bundle");
```
Кодировку забыли указать) К слову с ней что-то странное. Я сходу не смог добиться того, чтобы у меня русские буквы и в сообщениях и в вопросах нормально вывелись

BoLjshOJ committed:
HW 2
-fix all notice
1. Create IOService
2. Change @pathToFile form variable in bundle to forming string from example+locale
3. Update encoding

BoLjshOJ committed:
HW 2
-convert .csv to UTF-8
